src/server/analytic_services/prometheus_reports/base_prometheus_report.js

```diff
@@ -0,0 +1,44 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const prom_client = require('prom-client');
+const config = require('../../../../config.js');
+
+function is_enabled() {
+    return config.PROMETHEUS_ENABLED;
+}
+
+class BasePrometheusReport {
+    constructor() {
+        const client = this.prom_client;
```

Comment:
What am I reading here. This line make any sense?

Reply:
I am localizing the client into a local var for 2 reasons:
1. it is going to be accessed in a loop immediately after. 
2 it is going to be used with the `new` keyword which makes the code `new client[type]` much more readable than `new this.prom_client[type]` because it removes the ambiguity of what is being "newed"

Reply:
I don't see any loop after...

Reply:
You are correct. I was thinking about `noobaa_endpoint_report.js`. I will remove

Reply:
done